### ISSUE TITLE
Blacklist doesn't work

### ISSUE THREAD

**Issue by WenXin20** [skyjay1/GreekFantasy#55]
`Minecraft v1.16.5`
`Forge v36.1.16`
`Greek Fantasy v16.4.17`

The blacklist to prevent structures from spawning in certain biomes may not be working correctly. 
This is the current setup:
```
[features.satyr_camp]
	#Range: 0 ~ 1000
	chance = 15
	whitelist = false
	biome_types = ["NETHER", "END", "WATER", "COLD", "SNOWY", "MUSHROOM", "SANDY"]
```
But I am still finding structures spawning in mushroom and desert/beach biomes.

Also, Cerestas has z-fighting on it's teeth and Satyrs will stand on campfires and take damage constantly.
![2021-05-15_14 54 25](https://user-images.githubusercontent.com/12000593/118376431-7e5d6c00-b58d-11eb-8a23-7d64eba1d131.png)

**Comment by judicius** [comment]
I'll add to this that the Mob Spawns Dimension Whitelist doesn't seem to be working either 

`[mob_spawns]
	whitelist_dimensions = true
	dimensions = ["minecraft:overworld", "minecraft:the_nether"]
`
![2021-07-02_22 00 15](https://user-images.githubusercontent.com/30104638/124340117-e6d5cc00-db80-11eb-9706-0d36d85e1aee.png)

Ara spawning in the dimension Atum:Atum

**Comment by Hequinoxe** [comment]
I tried to reduce the chances for these two (Ara & Satyr) both for camp and creatures and it doesn't seem to work either. I also tried to change centaur chances to 800 then to 1 then to 1000 and I never saw any centaurs (but I saw them before I messed with the chances). I also tried to change the dryad (these seem to be working) and the nayad (chances seemed to work but when I tried to add River to the biomes I never saw any in rivers). Each and every time I made a modification I created a new world to test my changes.
